main.c

```diff
@@ -0,0 +1,300 @@
+#include "main.h"
+
+struct Node* CreateNode ()
+{
+  struct Node* newNode = (struct Node*)malloc(sizeof(struct Node));
+  if (!newNode) {
+    printf("Memory didn't allocated, cannot continue\n");
+    cleanup();
```

Comment:
What if a user does "malloc" for something of his own and then calls "CreateNode"? You'll exit the program without giving him a chance to call "free", please write a container that you yourself could potentially use in a bigger program.